### DISCUSSION TITLE
Error saving datasource (PostgreSQL)

### DISCUSSION THREAD

**Discussion by starmomou** [Budibase/budibase#3384] [Help] [answered]
Hi everyone, 
I created a table called `staffinfo` in database `lilymou`

<img width="1240" alt="螢幕截圖 2021-11-15 下午6 00 02" src="https://user-images.githubusercontent.com/73826858/141763419-81b7ab30-687a-44df-852e-e5d590a67926.png">

It showed an error message when I wanted to connect to PostgreSQL

<img width="357" alt="螢幕截圖 2021-11-15 下午5 37 06" src="https://user-images.githubusercontent.com/73826858/141763973-b0593f96-16e1-4eb6-8765-adf07d125fae.png">

I have tried so many times with different `Host` and `User` but all failed. How to resolve this problem? Thanks

<img width="1264" alt="螢幕截圖 2021-11-15 下午6 18 08" src="https://user-images.githubusercontent.com/73826858/141764992-0260885a-8d15-4b2d-9093-0355d1fd902c.png">

**Comment by mjashanks** [comment, marked as answer]
For your `Host`, use `172.17.0.1` if on linux, or `host.docker.internal` on windows and mac.

This is happening because Budibase is running inside a docker container - thus `localhost` is the address of the container, not the host machine.

**Reply by starmomou** [reply]
I've tried to use `host.docker.internal` as `Host` but it provided an error (I'm using cloud service)
<img width="1045" alt="螢幕截圖 2021-11-15 下午6 32 46" src="https://user-images.githubusercontent.com/73826858/141766603-0e9c78b0-f081-42f7-b801-5204e4e9f130.png">

**Reply by mjashanks** [reply]
Ah sorry, I assumed that you were using self-host, because your URL was "localhost".

`localhost` will not be accessible from the cloud service - as it's running on your own machine and Budibase cloud doe not have access to your machine. To use an external datasource with Budibase cloud, you will need a publically facing SQL database.

For example, you could set one up on Google Cloud or AWS.

**Reply by starmomou** [reply]
Thanks. I would try to use self-hosted budibase later
